engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/variables/GlobalVariablesPreparer.scala

```diff
@@ -22,15 +24,18 @@ class GlobalVariablesPreparer(userDefinedGlobalVariables: Map[String, ObjectWith
     prepareGlobalVariables(metaData).mapValuesNow(_.typ)
   )
 
+  private def toGlobalVariable(objectWithMethodDef: ObjectWithMethodDef, metaData: MetaData): ObjectWithType = {
+    objectWithMethodDef.obj match {
```

Comment:
I think if we could make this ObjectWithMethodDef return just ObjectWithType, and we would need this cast?

Reply:
I think we've decided to leave two separate method for value and return type. WDYT?

Reply:
well... I don't particularly like those pattern matchings all over the place... but let's leave this for a moment...